engine.go

```diff
@@ -446,6 +446,8 @@ func (e *Engine) QueryWithBindings(ctx *sql.Context, query string, parsed sqlpar
 
 		return nil, nil, nil, err
 	}
+
+	iter = rowexec.AddTransactionCommittingIter(iter, qFlags)
```

Comment:
if we keep doing this, it might make sense to group into a helper function with a doc comment